### PR TITLE
[REST] Check max file size of uploaded resource and extra resources in batch creation

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

Check the uploaded resource files when creating batch via REST API
- add config `kyuubi.batch.resource.file.max.size` for resource file's max size in bytes
- add config `kyuubi.batch.extra.resource.file.max.size` for each extra resource file's max size in bytes

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by wForget** [comment]
Should we limit uploaded file size in `writeToTempFile`?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -1876,6 +1876,26 @@ object KyuubiConf {
...
 
+  val BATCH_RESOURCE_FILE_MAX_SIZE: ConfigEntry[Long] =
+    buildConf("kyuubi.batch.resource.file.max.size")
+      .doc("The maximum size in bytes of the resource file" +
+        " that can be uploaded in a batch creating request via REST API." +
+        " 0 or negative value means no limit.")
+      .version("1.10.0")
+      .serverOnly
+      .longConf
+      .createWithDefaultString("100000000")
```
[the remark this answers is not in the document]
Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: Fixed, thx.

**Comment by bowenliang123** [comment]
> Should we limit uploaded file size in `writeToTempFile`?

Maybe no. The `writeToTempFile` method handles an unbounded stream with no predefined or known byte size. The file size is checked afterwards for varying purpose.

**Comment by wForget** [comment]
> > Should we limit uploaded file size in `writeToTempFile`?
> 
> Maybe no. The `writeToTempFile` method handles an unbounded stream with no predefined or known byte size. The file size is checked afterwards for varying purpose.

I see, thank you.

**Review by wForget — APPROVED** [review]
LGTM

**Pull request closed without merging**